### PR TITLE
Avoid overwriting bucket name if set by config

### PR DESCRIPTION
**Pull request by shayancanonical**
Resolves [issue 5](https://github.com/canonical/s3-integrator/issues/5)

# Issue
- The `bucket` config value is overwritten with the relation name when credentials are requested by the related application charm.
- The interface name for this charm should be `s3` and not `s3-credentials`
 
# Solution
- Avoid overwriting the bucket value if it is set by the config.
- Update interface name to `s3`

# Release Notes
Avoid overwriting bucket name if set by config

### CODE REVIEW

**Comment by deusebio** [comment]
Looks very good! 

Earlier today, I just wanted to make sure that this change was compliant with the s3 interface [here](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/s3/v0/README.md)

Indeed, the requirer's behavours requirements reads:

> Is expected to tolerate that the Provider may ignore the bucket field in some cases (e.g. S3Proxy or S3 Integrator) and instead use the bucket name received.

Which is what we are doing here. So green light to be merged!

However, while doing this, I realized that the name of the interface we are using is not "s3-credentails" (as in the metadata) but it should be "s3". I suppose we should really change this. We could either change it in this PR (if it is really one liner) or open a dedicated issue.

Up to you @shayancanonical ! I approve the PR, but if this is not addressed in the PR, just please open an issue to be addressed on a dedicated PR.

**Review by deusebio — APPROVED** [review]

**Comment by shayancanonical** [comment]
@deusebio @MiaAltieri as suggested, i renamed the interface from `s3-credentials` to `s3` in [3ec051e](https://github.com/canonical/s3-integrator/pull/9/commits/3ec051e03834ae8270baa3095b4788dbaede23dc). please update the consumer charms (mongodb) to use this new interface name once this PR is merged and this charm is publsihed

**Pull request merged**
